django_tenants_celery_beat/admin.py

```diff
@@ -12,9 +12,27 @@ class PeriodicTaskTenantLinkInline(admin.StackedInline):
     model = PeriodicTaskTenantLink
     can_delete = False
 
+    def get_formset(self, request, obj=None, **kwargs):
+        formset = super().get_formset(request, obj, **kwargs)
+        if obj is None:
+            # Only for adding a new PeriodicTask
+            formset.form.base_fields["tenant"].initial = request.tenant
+            if request.tenant.schema_name != get_public_schema_name():
+                # Hide all other Tenants
+                # Need to make the field non-readonly as otherwise the default value
+                # is blank, and the PeriodicTask will be created with no Tenant, which
+                # means it is then lost to the public schema
+                formset.form.base_fields["tenant"].queryset = (
+                    get_tenant_model().objects.filter(pk=request.tenant.pk)
+                )
+        return formset
+
     def get_readonly_fields(self, request, obj=None):
         if request.tenant.schema_name == get_public_schema_name():
             return tuple()
+        if obj is None:
+            # For new PeriodicTasks, we need to set the Tenant
+            return tuple()
```

Comment:
Could combine these? Or do you want the comment directly below the `obj is None` condition
```suggestion
        if request.tenant.schema_name == get_public_schema_name() or obj is None:
            # For new PeriodicTasks, we need to set the Tenant
            return tuple()
```

Reply:
Yeah, that's why I left them separate. I find it a bit clearer.